lib/tasks/one_off.rake

```diff
@@ -1,4 +1,20 @@
 namespace :pvb do
+  desc 'Merge Isle Of Wight Prisons'
+  task merge_iow: :environment do
+    albany    = Estate.find_by!(nomis_id: 'ALI').prisons.first
+    parkhurst = Estate.find_by!(nomis_id: 'IWI').prisons.first
+    FeedbackSubmission.where(prison_id: albany.id).
+      find_in_batches do |feebback_submissions|
+
+      FeedbackSubmission.where(id: feebback_submissions.map(&:id)).
+        update_all(prison_id: parkhurst.id)
+    end
+
+    Visit.where(prison_id: albany.id).find_in_batches do |visits|
+      Visit.where(id: visits.map(&:id)).update_all(prison_id: parkhurst.id)
+    end
+  end
```

Comment:
You misunderstood me, I wasn't talking about validations.

It's about ending with a partial update after running the task. The code is looping through the records using a condition, and at the same time updating the database that relate to the batch query condition.

When doing this in the past I've seen cases where only half the set of records to be updated got updated because batches get skipped as the dataset gets mutated.